packages/relay/src/TransactionManager.ts

```diff
@@ -375,6 +378,11 @@ data                     | ${transaction.data}
       return boostedTransactions
     }
 
+    // Sanity-check: the oldest tx.nonce immediately follows 'latest' nonce
+    if (nonce !== oldestPendingTx.nonce) {
```

Comment:
I think this is very dangerous as there are a lot of ways this can break. For one, I am not even sure if `oldestPendingTx` will be guaranteed to be `pending` here, I don't see what prevents it from being "unconfirmed", therefore being counted towards a nonce.

Reply:
@forshtat - I think its ok. what else would you expect ? 
By definition, if last mined TX is "n", then we must boost (or re-submit) TX "n+1"